libcore/src/androidTest/java/com/mapbox/android/core/crashreporter/MapboxUncaughtExceptionHanlderInstrumentationTest.java

```diff
@@ -65,6 +60,28 @@ public void testReportCap() throws IOException {
     assertEquals(2, files.length);
   }
 
+  @Test
+  public void ensureDirectoryWritableHasNoEffect() throws IOException {
+    writeToDisk(3);
+    MapboxUncaughtExceptionHanlder.ensureDirectoryWritable(context, TELEM_MAPBOX_PACKAGE);
+    assertEquals(3, directory.listFiles().length);
+  }
+
+  @Test
+  public void ensureDirectoryWritableCleansUp() throws IOException {
+    writeToDisk(10);
+    MapboxUncaughtExceptionHanlder.ensureDirectoryWritable(context, TELEM_MAPBOX_PACKAGE);
+    assertEquals(1, directory.listFiles().length);
```

Comment:
Shell we check the left one is the latest one?

Reply:
good point - i'll add that test to the next PR.